### PR TITLE
Cherry pick #740 to master

### PR DESCRIPTION
**Pull request by pleku**
Do not squash..

### CODE REVIEW

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/VaadinApplicationConfiguration.java [hunk 1]
```diff
@@ -36,18 +31,6 @@
...
 
-    /**
-     * Creates a default {@link ApplicationConfigurationFactory} bean in case
-     * there is no developer provided bean.
-     * 
-     * @return the default application configuration factory
-     */
-    @Bean
-    @ConditionalOnMissingBean
-    public ApplicationConfigurationFactory defaultApplicationConfigurationFactory() {
```
Comment (denis-anisimov) [inline vaadin-spring/src/main/java/com/vaadin/flow/spring/VaadinApplicationConfiguration.java:47]: This should not be removed

**Review by denis-anisimov — DISMISSED** [review]

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/VaadinApplicationConfiguration.java (hunk 1 above)
Comment (pleku) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/VaadinApplicationConfiguration.java]: Done

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/VaadinApplicationConfiguration.java [hunk 2]
```diff
@@ -25,9 +25,6 @@
 
 /**
  * Vaadin Application Spring configuration.
- * <p>
```
Comment (denis-anisimov) [inline vaadin-spring/src/main/java/com/vaadin/flow/spring/VaadinApplicationConfiguration.java:28]: This should not be removed as well.

Comment (pleku) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/VaadinApplicationConfiguration.java]: Done

**Review by caalador — APPROVED** [review]

**Comment by mike-goetz** [comment]
Hi, I have just upgraded my V14 MVC Application using vaadin-spring 17.0.1
Somehow it still is not working for me. 

Using 12.4.0, I have a second call to the method. In the first call, the wrapper is null, with the second call the wrapper was defined (triggered by context.refresh() from VaadinMVCWebAppInitializer

**Comment by pleku** [comment]
@mike-goetz I take it that you also updated the flow dependencies and not just the Spring add-on ? As the add-on version 17 requires Flow 7.0+ meaning that you bump to Vaadin 20.0. 

Please open a new issue to this project with the code+steps to reproduce the issue. Unfortunately we don't have a skeleton project for Vaadin+SpringMVC to make it easier.

**Pull request merged**